File: easy/1979-find-greatest-common-divisor-of-array/solution.java
/**
 * 1979. Find Greatest Common Divisor of Array
 * Time: O(n + log V)
 * Space: O(1)
 */
class Solution {
    public int findGCD(int[] nums) {
        int min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
        for (int x : nums) {
            min = Math.min(min, x);
            max = Math.max(max, x);
        }
        return gcd(min, max);
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }
}

File: easy/1979-find-greatest-common-divisor-of-array/solution.py
from typing import List
from math import gcd


class Solution:
    def findGCD(self, nums: List[int]) -> int:
        """
        1979. Find Greatest Common Divisor of Array
        Time: O(n + log V)
        Space: O(1)
        """
        return gcd(min(nums), max(nums))
